src/browser-installer/utils.ts

```diff
@@ -0,0 +1,187 @@
+
+export const browserInstallerDebug = debug("testplane:browser-installer");
+
+export const Browser = {
+    CHROME: PuppeteerBrowser.CHROME,
+    CHROMIUM: PuppeteerBrowser.CHROMIUM,
+    FIREFOX: PuppeteerBrowser.FIREFOX,
+    SAFARI: "safari",
+    EDGE: "MicrosoftEdge",
+} as const;
+
+export const Driver = {
+    CHROMEDRIVER: PuppeteerBrowser.CHROMEDRIVER,
+    GECKODRIVER: "geckodriver",
+    SAFARIDRIVER: "safaridriver",
+    EDGEDRIVER: "edgedriver",
+} as const;
+
+export type SupportedBrowser = (typeof Browser)[keyof typeof Browser];
+export type SupportedDriver = (typeof Driver)[keyof typeof Driver];
+
+export const getDriverNameForBrowserName = (browserName: SupportedBrowser): SupportedDriver | null => {
+    if (browserName === Browser.CHROME || browserName === Browser.CHROMIUM) {
+        return Driver.CHROMEDRIVER;
+    }
+
+    if (browserName === Browser.FIREFOX) {
+        return Driver.GECKODRIVER;
+    }
+
+    if (browserName === Browser.SAFARI) {
+        return Driver.SAFARIDRIVER;
+    }
+
+    if (browserName === Browser.EDGE) {
+        return Driver.EDGEDRIVER;
+    }
+
+    return null;
+};
+
+export const createBrowserLabel = (browserName: string, version = "latest"): string => browserName + "@" + version;
+
+export const getMilestone = (version: string | number): string => {
+    if (typeof version === "number") {
+        return String(version);
+    }
+
+    return version.split(".")[0];
+};
+
+export const semverVersionsComparator = (a: string, b: string): number => {
+    const versionPartsA = a
+        .replaceAll(/[^\d.]/g, "")
+        .split(".")
+        .filter(Boolean)
+        .map(Number);
+    const versionPartsB = b
+        .replaceAll(/[^\d.]/g, "")
+        .split(".")
+        .filter(Boolean)
+        .map(Number);
+
+    for (let i = 0; i < Math.min(versionPartsA.length, versionPartsB.length); i++) {
+        if (versionPartsA[i] !== versionPartsB[i]) {
+            return versionPartsA[i] - versionPartsB[i];
+        }
+    }
+
+    return 0;
+};
+
+export const normalizeChromeVersion = (version: string): string => {
+    const versionParts = version.split(".").filter(Boolean);
+
+    if (versionParts.length === 2) {
+        return versionParts[0];
+    }
+
+    if (versionParts.length >= 3) {
+        return versionParts.slice(0, 3).join(".");
+    }
+
+    return getMilestone(version);
+};
+
+export const getBrowserPlatform = (): BrowserPlatform => {
+    const platform = detectBrowserPlatform();
+
+    if (!platform) {
+        throw new Error(`Got an error while trying to download browsers: platform "${platform}" is not supported`);
+    }
+
+    return platform;
+};
+
+export const getChromePlatform = (version: string): BrowserPlatform => {
+    const milestone = getMilestone(version);
+    const platform = getBrowserPlatform();
+
+    if (platform === BrowserPlatform.MAC_ARM && Number(milestone) < MIN_CHROMIUM_MAC_ARM_VERSION) {
+        return BrowserPlatform.MAC;
+    }
```

Comment:
for chrome, use `mac64` platform, if chromium version is old and `mac_arm` version does not exist